openmetadata-ui/src/main/resources/ui/src/components/app-bar/Appbar.tsx

```diff
@@ -211,6 +211,19 @@ const Appbar: React.FC = (): JSX.Element => {
       );
     }
   };
+  const handleOnclick = (value: string) => {
+    setIsOpen(false);
+    addToRecentSearched(value);
+    history.push(
+      getExplorePathWithSearch(
+        value,
+        // this is for if user is searching from another page
+        location.pathname.startsWith(ROUTES.EXPLORE)
+          ? appState.explorePageTab
+          : 'tables'
+      )
+    );
```

Comment:
These line should be in a separate method and that method should be called from click as well as keydown(Enter) handler